db/report.go

```diff
@@ -25,25 +26,41 @@ func Report(id uint64, board, reason, ip string, illegal bool) error {
 	return err
 }
 
-// GetReports reads reports for a specific board. Pass "all" for global reports.
-func GetReports(board string) (rep []auth.Report, err error) {
+// GetReports reads reports and optionally, only illegal reports for a specific board.
+// Pass "all" for global reports.
+func GetReports(board string, illegal bool) (rep []auth.Report, err error) {
+	var where string
+	rep = make([]auth.Report, 0, 64)
 	tmp := auth.Report{
 		Board: board,
 	}
-	rep = make([]auth.Report, 0, 64)
+
+	if illegal {
+		if board == "all" {
+			where = "illegal = true"
+		} else {
+			where = fmt.Sprintf("board = '%s' and illegal = true", board)
```

Comment:
Is this branch ever actually called?

Reply:
Not currently, however it could be used for a staff reports live popup, or notification.

Reply:
If it is not used currently, remove it. Don't plan for contingencies too much. It leads to messy code.